server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java

```diff
@@ -86,35 +82,140 @@ public void destroy() {
     // Nothing to do
   }
 
-  static class ServletRequest extends org.sonar.server.ws.ServletRequest {
+  static class RestServletRequest extends org.sonar.server.ws.ServletRequest {
 
-    private final String path;
-    private final Map<String, String> additionalParams;
+    private final Response restResponse;
 
-    public ServletRequest(HttpServletRequest source, String path, Map<String, String> additionalParams) {
-      super(source);
-      this.path = path;
-      this.additionalParams = additionalParams;
+    public RestServletRequest(HttpServletRequest request) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Rename this class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2176)